racetrack_commons/racetrack_commons/deploy/job_type.py

```diff
@@ -93,22 +95,24 @@ def gather_job_types(
                 else:
                     raise RuntimeError(f'Invalid job type data. It should be list[tuple[Path, Path]], was {type(job_data)}')
 
-                _validate_dockerfile_paths(base_image_paths, template_paths, job_full_name)
-
                 name_parts = job_full_name.split(':')
                 assert len(name_parts) == 2, f'job type {job_full_name} should have the version defined (name:version)'
                 lang_name = name_parts[0]
                 lang_version = name_parts[1]
-                job_type_version = JobType(lang_name, lang_version, base_image_paths, template_paths)
+                job_type_version = JobType(
+                    lang_name=lang_name,
+                    version=lang_version,
+                    base_image_paths=base_image_paths,
+                    template_paths=template_paths,
+                )
                 job_types[job_full_name] = job_type_version
 
     return job_types
 
 
-def list_available_job_types(plugin_engine: PluginEngine) -> list[str]:
-    with wrap_context('gathering available job types'):
-        job_types = gather_job_types(plugin_engine)
-    return sorted(job_types.keys())
+def _validate_job_type(job_type: JobType):
+    job_full_name = f'{job_type.lang_name}:{job_type.version}'
```

Comment:
It looks like a candidate for a property in JobType: `def job_full_name(): return ....`, because probably this kind of code is repeated in other places.